### PR TITLE
[HF] Adding getters and const-correctness

### PR DESCRIPTION
**Pull request by hageboeck**
First part of PR #5324, which can be split off, because it's paving the way for future work.

### CODE REVIEW

File: roofit/histfactory/inc/RooStats/HistFactory/Data.h [hunk 1]
```diff
@@ -30,23 +30,24 @@ class Data {
   /// constructor from name, file and path. Name of the histogram should not include the path
   Data( std::string HistoName, std::string InputFile, std::string HistoPath="" );
 
-  std::string GetName() { return fName; }
+  std::string GetName() const { return fName; }
```
Comment (lmoneta) [inline roofit/histfactory/inc/RooStats/HistFactory/Data.h:33]: It is not maybe better to  return here and similar below a const reference to std::string ?

Comment (hageboeck) [reply roofit/histfactory/inc/RooStats/HistFactory/Data.h:33]: Better for sure, but this didn't do any harm in the past. The real CPU time is spent getting all histograms from files and converting them to likelihood models.

File: roofit/histfactory/inc/RooStats/HistFactory/FlexibleInterpVar.h [hunk 2]
```diff
@@ -56,6 +56,11 @@ namespace HistFactory{
     virtual void printMultiline(std::ostream& os, Int_t contents, Bool_t verbose = kFALSE, TString indent = "") const;
     virtual void printFlexibleInterpVars(std::ostream& os) const;
 
+    const RooListProxy& variables() const;
+    double nominal() const;        
+    const std::vector<double>& low() const;
+    const std::vector<double>& high() const;
```
Comment (hageboeck) [inline roofit/histfactory/inc/RooStats/HistFactory/FlexibleInterpVar.h:62]: These clash with local variables in the implementation, e.g. FlexibleInterpVar.cxx:56

Comment (hageboeck) [reply roofit/histfactory/inc/RooStats/HistFactory/FlexibleInterpVar.h:62]: and the ones in lines 91, 142

**Comment by hageboeck** [comment]
@phsft-bot build

**Pull request merged**